reactive/layer_gitlab.py

```diff
@@ -194,3 +198,34 @@ def configure_proxy():
 
     hookenv.status_set("active", HEALTHY)
     set_flag("reverseproxy.configured")
+
+
+def get_runner_token():
+    """Get the runner token for registering a runner."""
+    cmd = (
+        "gitlab-rails runner -e production "
+        "'puts Gitlab::CurrentSettings.current_application_settings.runners_registration_token'"
+    )
+    result = subprocess.check_output(cmd, shell=True)
```

Comment:
We don't need shell=True here if we just use the full path to gitlab-rails, and specify the command and arguments as an array - shell=True is discouraged from a security standpoint.

So something like -
```
cmd = [
    "/opt/gitlab/bin/gitlab-rails",
    "runner",
    "-e",
    "production",
    "'puts Gitlab::CurrentSettings.current_application_settings.runners_registration_token'",
]
token = subprocess.check_output(cmd, shell=True)
return token.decode("utf-8").rstrip()
```

Reply:
So i tried this previously, and I don't know if this is breaking down because of escaping in that long string arg or if it's something to do with calling rails from a sub-process w/o a shell. In either case, the only way I could get the actual key was to use the shell as all I get w/o a shell is an empty string from the command.

I'm testing the other change and I'll try your command to see if that works. Happy to not use shell if we can, I eventually decided this must be a case of shell required for rails to work. If you want to try this out, you don't need the charm at all, you can just hit a python env on your gitlab and run this command and see if you get a key.

Reply:
I've just re-tested, and indeed this is the same as what I tried originally. Token in this case (with the shell removed) returns: b''

I suspect this call is calling other file paths requiring a shell for it to process and return correctly, although I don't have any real evidence of that.